File: php/src/SharedKernel/Application/CommandBus/CommandBus.php
<?php

declare(strict_types=1);

namespace SharedKernel\Application\CommandBus;

use SharedKernel\Application\Command\Command;
use SharedKernel\Application\Command\Handler\Handler;

interface CommandBus
{
    public function registerHandler(Command $command, Handler $handler): void;
    public function handle(Command $command): void;
}

File: python/src/sharedkernel/application/commandbus/CommandBus.py
from abc import ABC, abstractmethod

from python.src.sharedkernel.application.command.Command import Command
from python.src.sharedkernel.application.command.handler.Handler import Handler


class CommandBus(ABC):

    @abstractmethod
    def register_handler(self, command: Command, handler: Handler):
        pass

    @abstractmethod
    def handle(self, command: Command):
        pass
